--- a/custom_components/homekit_controller/light.py
+++ b/custom_components/homekit_controller/light.py
@@ -201,3 +201,13 @@
         if hs_color is not None:
             characteristics[CharacteristicsTypes.HUE] = hs_color[0]
             characteristics[CharacteristicsTypes.SATURATION] = hs_color[1]
 
-        characteristics[CharacteristicsTypes.ON] = True
+        # Extract the manufacturer from the accessory associated with the service
+        manufacturer = (
+            getattr(self.service.accessory, "manufacturer", "")
+            if self.service and self.service.accessory
+            else ""
+        )
+
+        # VIMAR WORKAROUND: Omit ON command if sending other characteristics
+        if not characteristics or manufacturer != "Vimar":
+            characteristics[CharacteristicsTypes.ON] = True
 
         await self.async_put_characteristics(characteristics)
